src/sentry/http.py

```diff
@@ -69,22 +62,15 @@ def safe_urlopen(url, data=None, headers=DEFAULT_HEADERS,
     if not is_valid_url(url):
         raise SuspiciousOperation('%s matches the URL blacklist' % (url,))
 
-    req = urllib2.Request(url, data)
-    req.add_header('User-Agent', user_agent)
-    for key, value in headers:
-        req.add_header(key, value)
-
-    handlers = []
-    if not allow_redirects:
-        handlers.append(NoRedirectionHandler)
-
-    opener = urllib2.build_opener(*handlers)
+    all_headers = {header[0]: header[1] for header in headers}
+    all_headers.update({'User-Agent': user_agent})
 
-    return opener.open(req, timeout=timeout)
+    return requests.get(url, headers=all_headers, data=data, stream=True,
+                        allow_redirects=allow_redirects, timeout=timeout)
 
 
 def safe_urlread(request):
```

Comment:
If we switch to requests, we can remove this code to gzip decode since it'll do that automatically. As long as `Accept-Encoding: gzip` is sent, it'll be fine.

And if we look at what requests does, it already does all of this for us: https://github.com/kennethreitz/requests/blob/master/requests/utils.py#L564

So this is definitely duplicating efforts now.